src/libstd/sys/windows/fs.rs

```diff
@@ -377,8 +377,11 @@ impl fmt::Debug for File {
     }
 }
 
-pub fn to_utf16(s: &Path) -> Vec<u16> {
-    s.as_os_str().encode_wide().chain(Some(0)).collect()
+pub fn to_u16s(s: &Path) -> io::Result<Vec<u16>> {
+    if s.as_os_str().to_inner().to_inner().iter().any(|&b| b == 0) {
```

Comment:
Could this check the u16 encoded version instead of the wtf-8 encoded version? The inner encoding may actually use a 0 when the u16 later becomes nonzero (I'm not sure).